analyses/snv-callers/lancet-wxs-tests/explore-tcga-pbta.Rmd

```diff
@@ -0,0 +1,591 @@
+
+```{r}
+set_up_pbta <- function(data.name) {
+  # Given the name of dataset in the PBTA database, set up the TCGA data. 
+
+  # Start up connection
+  con <- DBI::dbConnect(
+    RSQLite::SQLite(),
+    file.path(scratch_dir, "snv_db.sqlite")
+  )
+  # Connect to SQL
+  df <- dplyr::tbl(con, data.name) %>%
+    # Only keep the columns we want
+    dplyr::select(cols_to_keep) %>%
+    as.data.frame() %>%
+    # Tack on the metadata columns we want
+    dplyr::inner_join(
+      pbta_metadata %>%
+        dplyr::select(
+          Tumor_Sample_Barcode = Kids_First_Biospecimen_ID,
+          experimental_strategy, short_histology
+        )
+    ) 
+  return(df)
+}
+```
+
+Function for setting up TCGA data from the database. 
+This database was originally created by the [run consensus bash script for tcga](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/run_caller_consensus_analysis-tcga.sh)
+
+```{r}
+set_up_tcga <- function(data.name) {
+  # Given the name of dataset in the TCGA database, set up the TCGA data. 
+  
+  # Start up connection
+  con <- DBI::dbConnect(
+    RSQLite::SQLite(), 
+    file.path(scratch_dir, "tcga_snv_db.sqlite")
+    )
+
+  # Connect to SQL
+  df <- dplyr::tbl(con, data.name) %>%
+    # Only keep the columns we want
+    dplyr::select(cols_to_keep) %>%
+    as.data.frame() %>%
+    # Shorten the Tumor_Sample_Barcode so it matches
+    dplyr::mutate(Tumor_Sample_Barcode = substr(Tumor_Sample_Barcode, 0, 12)) %>%
+    # Tack on the metadata columns we want
+    dplyr::inner_join(
+      tcga_metadata %>% 
+        dplyr::select(
+          Tumor_Sample_Barcode, 
+          short_histology,
+          experimental_strategy
+          )
+      )
+
+  return(df)
+}
+```
+
+Function for making a combined CDF plot for TMB. 
+This plotting function was adapted from the [`breaks_cdf_plot` function in the 
+`chromosomal-instability`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/b1b73fe321a97fa82d85c86d20bd85635aabba25/analyses/chromosomal-instability/util/chr-break-plot.R#L120)
+
+```{r}
+tmb_cdf_plot <- function(tmb_df, plot_title) {
+  # Given a data.frame of TMB data, plot it as a CDF plot and save it as a png. 
+  #
+  # Args:
+  #   tmb_df: a chromosomal breaks density file path where each sample is
+  #                 a row with `samples` and `breaks_count` columns. 
+  #   plot_title: to be used for the ggplot2::ggtitle and what it will be saved 
+  #                as a png as.
+  
+  cdf_plot <- tmb_df %>%
+    as.data.frame() %>%
+    dplyr::mutate(short_histology = tools::toTitleCase(short_histology)) %>%
+    # Only plot histologies groups with more than `min_samples` number of samples
+    dplyr::group_by(short_histology, add = TRUE) %>%
+    # Only keep groups with this amount of samples
+    dplyr::filter(dplyr::n() > 5) %>%
+    # Calculate histology group mean
+    dplyr::mutate(
+      hist_mean = mean(tmb),
```

Comment:
median here too?

Reply:
I kind of think we should have this archived as we looked at originally since this doesn't go toward any plots we will use.

Reply:
But when I make this plot into a function I can make it median instead.